libs/fabric/src/lib/components/core/shared/changeable-item.directive.ts

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { EventEmitter, Input, Output } from '@angular/core';
+
+import { OnChanges, TypedChanges } from '../../../declarations/angular/typed-changes';
+import { ItemChangedPayload } from '../../core/declarative/item-changed.payload';
+import { ChangeableItemHelper } from './changeable-helper';
+
+/**
+ * Parent class for wrapper directive for single item with OnChanges
+ */
+export abstract class ChangeableItemDirective<TItem> implements OnChanges<ChangeableItemDirective<TItem>> {
+  @Input()
+  key: string;
+
+  @Output()
+  get onItemChanged(): EventEmitter<ItemChangedPayload<string, TItem>> {
+    return this.changeableItemHelper && this.changeableItemHelper.onItemChanged;
+  }
+
+  private changeableItemHelper: ChangeableItemHelper<TItem>;
+
+  ngOnInit() {
+    this.changeableItemHelper = new ChangeableItemHelper(this.key);
+  }
+
+  ngOnChanges(changes: TypedChanges<TItem>) {
+    if (this.changeableItemHelper) {
+      this.changeableItemHelper.onChanges(changes);
+    }
+  }
```

Comment:
Don't you also need to have `ngOnDestory()`? (calling `this.changeableItemHelper.onDestory()`)

Reply:
`this.changeableItemHelper` doesn't have an `onDestroy()` function since it doesn't have any Subscriptions (unlike ChangeableItemsHelper). Would an `onDestroy` still be needed for this case?

Reply:
Ah, no. My miss here.